modules/core/src/main/scala/com.snowplowanalytics/iglu.schemaddl/parquet/Migrations.scala

```diff
@@ -0,0 +1,205 @@
+package com.snowplowanalytics.iglu.schemaddl.parquet
+
+import cats.Show
+import cats.syntax.all._
+import com.snowplowanalytics.iglu.schemaddl.parquet.Type.{Array, Struct}
+
+/*
+  Parquet schemas are migrated though the `merge.schema = true` loading option in Spark/Databricks.
+  Migrations in this package do not define spark 3.0 style `ATLER TABLE` transformations, similar to Redshift. Instead
+  they are used to categorize schema changes into 2 categories:
+   - Non breaking - Miscellaneous changes that would be handled by the `merge.schema`:
+      * `NullableRequired` - nullable to required change
+      * `NestedKeyAddition` - addition of the new Struct key anywhere except for top level Schema.
+      * `TopLevelKeyAddition` - key addition in top level schema
+      * `KeyRemoval` historical data loss, such as removing field in Struct. Loading will succeed with data loss.
+      * `RequiredNullable` Nullable to required field or array migration. Loading will fail.
+   - Breaking - Breaking change that would either lead to oen of these outcomes:      
+      * `IncompatibleType` Type casting is not allowed with an exception for Struct fields.
+ */
+object Migrations {
+
+  // Path to field or type within the Parquet schema structure. String `"[arrayDown]"` used to indicate array traversal
+  type ParquetSchemaPath = List[String]
+
+
+  // Base trait for schema differences, which would result in migration.
+  sealed trait ParquetMigration {
+    // As Migration logic traverses deeper into the structure keys appended to the head of list.
+    // So path stored is reverse.
+    def reversedPath: ParquetSchemaPath
+
+    def path: ParquetSchemaPath = reversedPath.reverse
+  }
+
+  type ParquetSchemaMigrations = Set[ParquetMigration]
+
+  sealed trait NonBreaking extends ParquetMigration
+
+  sealed trait Breaking extends ParquetMigration
+
+  implicit val showPerson: Show[ParquetMigration] = Show.show(_.toString)
+
+  case class KeyRemoval(override val reversedPath: ParquetSchemaPath, removedKey: Type) extends NonBreaking {
+    override def toString: String = s"Key removal at /${path.mkString("/")}"
+  }
+
+  case class RequiredNullable(override val reversedPath: ParquetSchemaPath) extends NonBreaking {
+    override def toString: String = s"Changing nullable property to required at /${path.mkString("/")}"
+  }
+
+  case class NullableRequired(override val reversedPath: ParquetSchemaPath) extends NonBreaking {
+    override def toString: String = s"Changing required property to nullable at /${path.mkString("/")}"
+  }
+
+  case class TopLevelKeyAddition(override val reversedPath: ParquetSchemaPath, key: Type) extends NonBreaking {
+    override def toString: String = s"Top-level schema key addition at /${path.mkString("/")}"
+  }
+
+  case class NestedKeyAddition(override val reversedPath: ParquetSchemaPath, key: Type) extends NonBreaking {
+    override def toString: String = s"Nested object key addition at /${path.mkString("/")}"
+  }
```

Comment:
Is there a reason to make a distinction between top level and tested additions?  Can it be simply KeyAddition?

I think you needed this distinction in an earlier version of this PR but I don't see why it's needed now.